org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -94,8 +94,12 @@
 	@Inject 
 	private OperationCanceledManager operationCanceledManager;
 	
+	private ClosedProjectsQueue closedProjectsQueue;
+	
 	@Inject
-	private ReflectExtensions reflector;
+	private void injectClosedProjectsQueue(ISharedStateContributionRegistry sharedState) {
+		this.closedProjectsQueue = sharedState.getSingleContributedInstance(ClosedProjectsQueue.class);
```

Comment:
why is this needed here and not in ProjectOpenedOrClosedListener?

Reply:
The projectOpenOrClosedListener is instantiated from the same child injector. The build is from the parent injector.